### PR TITLE
STSMACOM-301 Create an Input Type = Text area

### PR DESCRIPTION
**Pull request by BogdanDenis**
## Purpose

- add a possibility to create custom fields with type Text Area

## Link

https://issues.folio.org/browse/STSMACOM-301

## Screenshots

ViewCustomFieldsSettings
Screenshot_4

EditCustomFieldsSettings
Screenshot_3

### CODE REVIEW

File: lib/CustomFields/components/FieldAccordion/view-sections/TextboxSection.js [hunk 1]
```diff
@@ -21,9 +21,18 @@ const TextboxSection = props => (
...
+            {props.helpText}
+          </KeyValue>
+        </Col>
+      ) : null
+    }
     <Col xs={3}>
-      <KeyValue label={<FormattedMessage id="stripes-smart-components.customFields.helperText" />}>
-        {props.helpText || <NoValue />}
+      <KeyValue label={<FormattedMessage id="stripes-smart-components.customFields.required" />}>
+        {props.required ? 'Yes' : 'No'}
```
Comment (Vladyslav-Velytskyi) [inline lib/CustomFields/components/FieldAccordion/view-sections/TextboxSection.js]: Change `'Yes'`,  `'No'` to `<FormattedMessage id="..." />`, you need to add this strings to translations

File: translations/stripes-smart-components/ar.json [hunk 2]
```diff
@@ -178,7 +178,6 @@
     "reset": "إعادة تعيين",
     "notes.assignUnassignAll": "تعيين / إلغاء تعيين كل الملاحظات",
     "customFields": "حقول مخصصة",
-    "customFields.new": "جديد",
```
Comment (MaksymDryha) [inline translations/stripes-smart-components/ar.json]: You only have to make changes to en.json. All other files will be automatically updated by the translation bot after the PR is merged.

File: lib/CustomFields/components/FieldAccordion/view-sections/TextboxSection.js (hunk 1 above)
Comment (BogdanDenis) [reply lib/CustomFields/components/FieldAccordion/view-sections/TextboxSection.js]: Fixed in new commit

File: lib/CustomFields/components/FieldAccordion/FieldAccordion.js [hunk 3]
```diff
@@ -58,9 +60,6 @@ const FieldAccordion = (props) => {
       {required && !visible
         && <FormattedMessage id="stripes-smart-components.customFields.settings.accordion.requiredHidden" />
       }
-      {required && visible
-        && <FormattedMessage id="stripes-smart-components.customFields.settings.accordion.required" />
-      }
       {!visible && !required
         && <FormattedMessage id="stripes-smart-components.customFields.settings.accordion.hidden" />
       }
```
[the remark this answers is not in the document]
Comment (BogdanDenis) [reply lib/CustomFields/components/FieldAccordion/FieldAccordion.js]: Fixed in new commit

File: translations/stripes-smart-components/ar.json (hunk 2 above)
Comment (BogdanDenis) [reply translations/stripes-smart-components/ar.json]: Fixed in new commit

File: lib/CustomFields/components/FieldAccordion/view-sections/TextboxSection.js [hunk 4]
```diff
@@ -21,9 +21,22 @@ const TextboxSection = props => (
...
+      ) : null
+    }
     <Col xs={3}>
-      <KeyValue label={<FormattedMessage id="stripes-smart-components.customFields.helperText" />}>
-        {props.helpText || <NoValue />}
+      <KeyValue label={<FormattedMessage id="stripes-smart-components.customFields.required" />}>
+        {
+          props.required ? (
+            <FormattedMessage id="stripes-smart-components.customFields.required.checked" />
+          ) : <FormattedMessage id="stripes-smart-components.customFields.required.unchecked" />
```
Comment (MaksymDryha) [inline lib/CustomFields/components/FieldAccordion/view-sections/TextboxSection.js]: Please, format ternary operators like this: 
```jsx
props.required
  ? <FormattedMessage id="stripes-smart-components.customFields.required.checked" />
  : <FormattedMessage id="stripes-smart-components.customFields.required.unchecked" />
```

File: lib/CustomFields/components/FieldAccordion/view-sections/TextboxSection.js [hunk 5]
```diff
@@ -21,9 +21,22 @@ const TextboxSection = props => (
...
       </KeyValue>
     </Col>
+    {
+      props.helpText ? (
+        <Col xs={3}>
+          <KeyValue label={<FormattedMessage id="stripes-smart-components.customFields.helperText" />}>
+            {props.helpText}
+          </KeyValue>
+        </Col>
+      ) : null
```
Comment (MaksymDryha) [inline lib/CustomFields/components/FieldAccordion/view-sections/TextboxSection.js]: Same for ternary operator here

**Pull request merged**